### PR TITLE
Fix typehint error for asserting on response headers

### PR DESCRIPTION
**Pull request by NiR-**

### CODE REVIEW

File: src/Knp/FriendlyContexts/Context/ApiContext.php [hunk 1]
```diff
@@ -16,6 +17,23 @@ public function getResponse()
...
+    {
+        if (null !== $this->response) {
+            throw new \InvalidArgumentException('First, you need to send a request.');
+        }
+
+        $headers = $this->response->getHeaders();
+
+        if ($headers instanceof ToArrayInterface) {
+            return $headers->toArray();
+        } elseif (is_array($headers)) {
```
Comment (Einenlum) [inline src/Knp/FriendlyContexts/Context/ApiContext.php]: elseif is useless here, since there is a return before.

Comment (NiR-) [inline src/Knp/FriendlyContexts/Context/ApiContext.php]: Still there are 3 possible cases:
  * `getHeaders` returns an object implementing `ToArrayInterface`.
  * `getHeaders` returns an array.
  * `getHeaders` returns none of 2 first cases.

So the `elseif` is not useless at all.

Comment (Einenlum) [inline src/Knp/FriendlyContexts/Context/ApiContext.php]: I mean the second one could be an "if" and not an "elseif".

Comment (NiR-) [inline src/Knp/FriendlyContexts/Context/ApiContext.php]: Closed. If you want to fix this bug, I'm pretty sure you'll fix that `elseif` issue better than me. :)

**Pull request closed without merging**